client/src/lib/RenderReviewList.js

```diff
@@ -0,0 +1,219 @@
+import DataTable, {createTheme} from 'react-data-table-component';
+import React from 'react';
+import {format} from 'date-fns';
+import Select from 'react-select';
+
+const RenderReviewList = () => {
+    let lastFetch;
+    const queryString = new URLSearchParams(window.location.search);
+    const status = queryString.get('status') ?? 'pending';
+    const [data, setData] = React.useState([]);
+    const [loading, setLoading] = React.useState(false);
+    const [totalRows, setTotalRows] = React.useState(0);
+    const [perPage, setPerPage] = React.useState(10);
+    const [currentPage, setCurrentPage] = React.useState(1);
+    const [channelOptions, setChannelOptions] = React.useState([]);
+    const [selectedChannel, setSelectedChannel] = React.useState(queryString.get('channel') ?? 'all');
+    const [selectPlaceHolder, setSelectPlaceHolder] = React.useState('Code Reviews For All Slack Channels');
+
+    const fetchData = async ({channel, limit, page, status, description}) => {
+        // Check last fetch to avoid data refresh during first page loading.
+        if (lastFetch && lastFetch === `${channel}-${limit}-${page}-${status}`) {
+            return;
+        }
+        lastFetch = `${channel}-${limit}-${page}-${status}`;
```

Comment:
Have try that, also tried useRef.  Because the state won't update until the next react side affect trigger.  So the concurrent call to render will trigger multiple call because the variable is not update right away:
![Screenshot from 2023-02-28 13-43-55](https://user-images.githubusercontent.com/5245991/221988307-ebc4dfeb-644c-453b-8568-d592f8caa706.png)
For the page refresh, the fetchData got called 3x instead of one time.  This is because of the data table component trigger changes when it initializing that cause a few handle to fire at the same time during the same cycle of the same state.

eg. 
1. Page render, lastFetch = '', because React.useState('')
    - 1a. fetchData got trigger by useAffect, -> setLastFetch only update the state.  lastFetch still have empty value
    - 1b. page update handler got triggered, fetchData got call, lastFect = '', thus trigger another remote data fetch
    - 1c. page size update handler got triggered, fetchData got call, lastFect = '', thus trigger another remote data fetch

Reply:
I see what you're saying. This still feels like the wrong way for doing this. Could you check `loading === false` at the top of `fetchData` and only let it run if it's not currently executing a fetch? 

I think a more React way of doing this would be to do

```
// current channel, limit, page, status, description are all state variables

// You have handle functions that do this
const handlePageChange = (page) => {
  setCurrentPage(page)
}

const handlePerRowsChange = (newPerPage, page) => {
  setCurrentPage(page);
  setPerPage(newPerPage);
}

// and etc. with the other handle changes, then
React.useEffect(() => {
  fetchData({ channel, limit, page, status, description }); // these don't even need to be arguments any more, since they are in state, fetchData would know them
}, [channel, limit, page, status, description]);
```
then any time one of `channel, limit, page, status, description` changes then that useEffect will run the fetch.

Reply:
I don't think this refactor is blocking merging these highly anticipated updates :D but a refactor in the near future is very much worth the time and will prevent bugs.

Reply:
Check loading, it's the same because it reference useState.  The value return is a snapshot.  setLoading will not happen until the next cycle.  Unless we move the fetchData outside of the component function and implement as global, there is no other way around the duplicate calls.  Another word, the fetchData is just a helper function, not part of a state.  The fetchData var is part of the fetchData global tracking and not part of the component state value.